speechbrain/core.py

```diff
@@ -1273,7 +1273,7 @@ def on_fit_batch_end(self, batch, outputs, loss, should_step):
         """
         pass
 
-    @torch.no_grad()
+    @torch.inference_mode()
```

Comment:
@Adel-Moumen this breaks backward compatibility with older version of torch no?

Reply:
`@torch.inference_mode()` came in v1.9. If you want I can swap it with `@torch.no_grad()`.

Reply:
Yep swap it.

Reply:
Done.